File: 392-is-subsequence/392-is-subsequence.go
func isSubsequence(s string, t string) bool {
    
    i, j := 0, 0
    
    for i != len(s) && j < len(t) {
        if s[i] == t[j] {
            i = i+1
            j = j+1
        } else {
            j = j + 1
        }
    }
    
    return i == len(s)
    
}

File: 392-is-subsequence/392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        
        if len(s) > len(t):
            return False
        i, j = 0, 0
        m, n = len(s), len(t)
        
        while i < m and j < n:
            if s[i] == t[j]:
                i += 1
                j += 1
            else:
                j += 1
        return i == m